### ISSUE TITLE
[wf-panel] Lags after hiding

### ISSUE THREAD

**Issue by NamorNiradnug** [WayfireWM/wf-shell#189]
Seems like the window isn't being rerendered after being hidden and losing focus.

https://github.com/WayfireWM/wf-shell/assets/50048443/466fb206-0a0b-458d-8fa7-48fb62f89d7e

**Comment by ammen99** [comment]
Is this a regression?

**Comment by NamorNiradnug** [comment]
> Is this a regression?

Yes, it started to happen after I updated Wayfire to 0.8 from 0.7.5

**Comment by soreau** [comment]
My thoughts as well, maybe bisect? It could be related to [this recent commit](https://github.com/WayfireWM/wf-shell/commit/665263593350fddd92322e553fc09895cdebf209).

**Comment by NamorNiradnug** [comment]
> My thoughts as well, maybe bisect? It could be related to [this recent commit](https://github.com/WayfireWM/wf-shell/commit/665263593350fddd92322e553fc09895cdebf209).

I'm not sure, but I think it was happening before I merged this commit, let me check.

**Comment by NamorNiradnug** [comment]
> > My thoughts as well, maybe bisect? It could be related to [this recent commit](https://github.com/WayfireWM/wf-shell/commit/665263593350fddd92322e553fc09895cdebf209).
> 
> I'm not sure, but I think it was happening before I merged this commit, let me check.

Yes, it behaves like that with and without that commit.

**Comment by NamorNiradnug** [comment]
For some reason nothing happens after [this `queue_draw`](https://github.com/WayfireWM/wf-shell/blob/master/src/util/wf-autohide-window.cpp#L364) (at least [`draw_signal`](https://github.com/WayfireWM/wf-shell/blob/master/src/util/wf-autohide-window.cpp#L33-L34) isn't being emitted)

**Comment by soreau** [comment]
It should be easy to bisect if you can find a good commit. Have you checked WAYLAND_DEBUG output and make sure the exclusive zone is being set as you expect? Also, it could be that something changed on the wayfire side.

**Comment by NamorNiradnug** [comment]
So, I've found the commit which introduced the bug
That's this one commit: https://github.com/WayfireWM/wayfire/commit/7f8004b41a1865a984f251da3e156cd2d6247368

**Comment by NamorNiradnug** [comment]
Still as issue with Wayfire 0.9

**Comment by AKArien0** [comment]
Experiencing the same issue. Fix in #276 seems to work, but is jank as it sounds (a permanent white bar is at the edge of the screen. I’d assume it’s colour will change with the associated css ?).If someone can figure out how to fix this,

**Comment by AKArien0** [comment]
Experiencing the same issue. Fix in https://github.com/WayfireWM/wf-shell/issues/276 seems to work, big thanks for it, though it brings a permanent bar is at the edge of the screen.
If someone can figure out how to fix "properly" this, know you will have my eternal gratitude.

**Comment by ammen99** [comment]
I think the issue is likely in Wayfire, can someone try this patch https://0x0.st/8fpU.diff (it is for Wayfire)?

**Comment by NamorNiradnug** [comment]
Works for me!
